core/src/sumeragi/main_loop.rs

```diff
@@ -842,17 +875,19 @@ fn reset_state(
         // a view change a new block is immediately created by the leader
         *round_start_time = Instant::now();
         was_commit_or_view_change = true;
+        *old_view_change_index = 0;
```

Comment:
are you sure you want to reset this on every commit?

Reply:
Yes. View change index is a round local value.

Reply:
but it didn't used to be?

Reply:
I'm with @mversic here. We need to be extra careful with the consensus. Whatever is the response to this becomes a comment.

Reply:
I wrote the old code. I wrote this code. I assure you the behaviour is the same. When I say round, I mean what happens after committing block N and before committing N+1. Within a round 0-INF view changes can occure. It is important to keep track of this value so that we can shift the topology the correct number of times. When a block has been committed the view change index is now zero, then we increment it with each topology rotation.

Reply:
Appealing to authority isn't addressing the comment. 

If this isn't obvious to @mversic, a senior engineer, it's not obvious to whomever is going to read the code afterwards.